server/src/server.ts

```diff
@@ -324,6 +330,44 @@ connection.onCompletion((completionInfo: CompletionParams): CompletionItem[] =>
           const searchResult = findStatement(decl.body, null, { pos, genv: decls });
           if (searchResult === null || searchResult.environment === null) break;
 
+          const doc = documents.get(completionInfo.textDocument.uri);
+          if (doc) {
+            const context = getCompletionContext(
+              doc.getText(),
+              doc.offsetAt(completionInfo.position));
+
+            if (context) {
+              switch (context.tag) {
+                case CompletionContextKind.StructAccess:
+                  try {
+                    // Type safety? :D 
+                    const type = <ast.Type>synthExpression(decls, searchResult.environment, null, <ast.Expression>context.expr);
+                    const structname = getStructId(decls, type);
+                    
+                    const struct = getStructDefinition(decls, structname);
+                    if (struct && struct.definitions) {
+                      return struct.definitions.map(field => ({
+                          label: field.id.name,
+                          kind: CompletionItemKind.Field,
+                          documentation: mkMarkdownCode(`${typeToString(field.kind)} ${struct.id.name}::${field.id.name}`),
```

Comment:
Maybe don't use the :: notation since that's not taught to them. Just use the field name (like how it's defined in the struct, it's not "int foo::bar;" it's "int bar;"). Or something like this, open to discussion.

Reply:
How about showing something like this:

```
struct bar { 
  ... 
  int foo;
}
```

basically showing an excerpt from the struct declaration

Reply:
Yeah I like that. It's not as clean as the :: but it more closely matches what they already know with structs.